add Dataverse documentation

joel-simpson opened the pull request:
Adds Dataverse Integration page providing an overview of Archivematica to Dataverse integration. Update the Integrations page and Transfers page with relevant references / links

connected to https://github.com/artefactual/archivematica/issues/1086
connected to https://github.com/artefactual/archivematica/issues/1147

joel-simpson committed:
add Dataverse links

Adds Dataverse Integration section with links to Dataverse Integration page in Archivematica docs and dataverse section in Storage Service docs.

joel-simpson committed:
Create dataverse.rst

Adding a new Dataverse Integration page that provides an overview of how integration between the two systems work and what is unique about processing a Dataverse Dataset. Links to separate page in storage service for configuration instructions.

joel-simpson committed:
update transfer.rst with dataverse link

Add link to Transfer page with a 'see also' link to the Dataverse Integration page

joel-simpson committed:
fix anchor names

change anchor names to be unique

joel-simpson committed:
typo fix

fix typo - missing bracket

joel-simpson committed:
add dataverse to index

joel-simpson committed:
fix duplicate label

changed the label for this page from dataverse-integration to dataverse-transfer so that it is distinct from the dataverse-integration label used in the integrations page.

joel-simpson committed:
update label used for link to overview page

Updated link to the Dataverse integration page (to disambiguate from label used in SS docs)

joel-simpson committed:
typo in label

fixed typo of label from dataverse-transfer to dataverse-transfers

sallain reviewed (commented):
I think this looks good, @joel-simpson! @sevein noticed a couple of places where there's a period missing, so I'll add some comments to reflect that.

If you have time, can you squash the commits? It would be nice to merge this as one commit instead of 9.

sallain commented on admin-manual/installation-setup/integrations/integrations.rst:
```diff
@@ -50,4 +51,15 @@ Arkivum
+See :ref:`Dataverse integration <dataverse-transfers>` for an overview of how 
+Dataverse datasets are preserved. 
+For instructions to configure a Dataverse integration, see 
+:ref:`Storage Service docs <storageservice:administrators>`
```
Needs a period.

joel-simpson replied:
agreed

sallain commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+-------------
+
+Integration with a Dataverse repository is configured in the Storage Service. 
+For detailed instructions, see the :ref:`Administrators Manual <storageservice:administrators>`
```
Needs a period.

joel-simpson replied:
yep

sallain commented on user-manual/transfer/transfer.rst:
```diff
@@ -54,6 +54,9 @@ transfer(s). Once this is done, the SIP can be moved into ingest or sent to a
    :ref:`Forensic image processing <forensic>`.
+   
+   If your transfer is a Dataverse dataset, please see 
+   :ref:`Dataverse Integration <dataverse>`
```
Needs a period

joel-simpson committed:
fix typo

joel-simpson committed:
fix typo

joel-simpson committed:
fix typo

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+`Dataverse`_ is an open source repository for research data. Archivematica can
+be configured to use a Dataverse Repository as a Transfer Source Location. 
+Dataverse Transfer Source Locations can be configured to display all available 
+datasets or a subset using query parameters.
```
Suggested: `subset of them`.

**NB.** Depending on how a query is configured a user will only see the first 50 results returned.

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+Selecting Datasets for preservation
+-----------------------------------
+
+When a Dataverse Transfer Source Location is selected in the Transfer Tab of 
```
I'm not sure about the capitalization of 'Transfer Tab'

joel-simpson replied:
good spot. It looks like in the rest of the docs 'Transfer tab' is used so am changing to that.

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+* a <mets:fileSec> section that lists all files provided, grouped by type 
+  (original, metadata or derivative)
+
+* a <mets:structMap> section that describes the structure of the files as 
```
For snippets of xml like this, double inverted quotations are a good idea to make it clearer to read, e.g. ` ``<mets:fileSec>`` `.

![image](https://user-images.githubusercontent.com/1880412/45992984-21b28900-c052-11e8-92c8-7989a49864db.png)

joel-simpson replied:
thanks for the suggestion. Have updated these with the additional quotations.

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+directory icon will expand the view to display the list of files included in 
+the dataset. Individual files can’t be selected for Transfer.
+
+When a Dataverse dataset is selected, the Transfer Type ‘Dataverse’ must also 
```
Similarly, not sure about capitalization of transfer type.

joel-simpson replied:
On the Transfer page of our docs, neither is capatilized so have corrected to be consistent.

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+* a <mets:structMap> section that describes the structure of the files as 
+  provided by Dataverse (particularly helpful for understanding which files were
+  provided in `Tabular Data file bundle`_)
+
```
I think this is useful. There seems to be a few connecting words missing. Suggest:  `this is particularly helpful for understanding the files that are provided in a tabular data file 'bundle'`. 

But I'd also introduce a bundle somewhere else, maybe in describing the feature?

joel-simpson replied:
good suggestions. I updated the wording as suggested, and also added a section on dataset contents - to highlight that we retrieve all 'bundles'.

sallain replied:
Sorry for one more nitpick - there's a hanging ) in line 93

joel-simpson replied:
well spotted. now fixed.

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+  provided in `Tabular Data file bundle`_)
+
+The Dataverse METS is found in the final AIP in this location: 
+<AIP Name>/data/objects/metadata/transfers/<transfer name>/METS.xml 
```
Like above, path strings fit well inside double inverted commas ` ``path/to/something`` `.

joel-simpson replied:
thanks, now updated.

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+
+The Dataverse METS is found in the final AIP in this location: 
+<AIP Name>/data/objects/metadata/transfers/<transfer name>/METS.xml 
+(This is also where you will find the dataset.json metadata file provided by 
```
As we haven't introduced `dataset.json` we might change the definitive `the` for the indefinite `a`.

joel-simpson replied:
agreed

joel-simpson replied:
actually changed back to 'the' after adding a section describing dataset contents (where I now introduce the dataset.json file)

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+
+The Archival Information Package (AIP) METS file follows the basic structure 
+for a standard Archivematica AIP METS file. Derivatives generated by Dataverse
+are indicated using the METS fileGrp attribute (where USE =“derivative”).
```
Suggest ` ``USE="derivative"`` ` (double inverted commas). For the dmdSecs in brackets below as well.

joel-simpson replied:
yep

ross-spencer commented on user-manual/transfer/dataverse.rst:
```diff
@@ -0,0 +1,118 @@
+
+In the PREMIS Object entity, relationships between original and derivative 
+tabular format files from Dataverse are described using PREMIS relationship 
+semantic units. A PREMIS derivation event indicates the derivative file was 
```
Possibly just need `... PREMIS semantic units.`.

joel-simpson replied:
yes, that does read better. (less is more)

ross-spencer commented on user-manual/transfer/transfer.rst:
```diff
@@ -54,6 +54,9 @@ transfer(s). Once this is done, the SIP can be moved into ingest or sent to a
    :ref:`Forensic image processing <forensic>`.
+   
+   If your transfer is a Dataverse dataset, please see 
+   :ref:`Dataverse Integration <dataverse>`.
```
Dataset seems a little diminutive, although I appreciate that is technically what we're requesting from the API. We've seen fairly significant transfers from just our demo site. 

I would suggest the text:

`If your transfer is from a Dataverse repository, please see ...`

joel-simpson replied:
that's interesting... I actually think of many datasets being massive (e.g. astronomy examples I've heard of that are TB big). But your wording suggestion works just as well so have updated to that.

ross-spencer replied:
Good point! One of the first presentations I saw in this field was CERN. They had a little bit of data I suppose ;)

ross-spencer reviewed (changes requested):
Hi @joel-simpson. This is pretty good. The main things I have spotted are in your placement of xml snippets or OS paths inline. These display a little better using double inverted commas. There are a few other small suggestions I've made along the way.

joel-simpson committed:
add section on bundles

Updated various typos / formatting suggestions from Ross, and added a section to describe the contents of a dataset (and bundles).

joel-simpson committed:
minor wording update

joel-simpson committed:
fix typo

joel-simpson committed:
fix typo

sallain commented:
Thanks Joel! This has been merged into 1.8 [here](https://github.com/artefactual/archivematica-docs/commit/20dd22faa5438cd9557a8e948749abb751f4cdf9).
